### PR TITLE
Feature/15-react-zustand

### LINKED ISSUE #15: React + Zustand 뼈대 만들기

### PR DESCRIPTION
**Pull request by sprinter707**
📖 스토리
https://github.com/ga0hyeon/CN-SIDE-1/issues/15

💻 작업내용
- React + Zustand 틀 구성
- 예시 코드 추가 (Ui-components 패키지에서 hook 생성 후 chromeside에서 호출)
- 패키지 정리
  - chromeside-board (메인 로비)
  - ui-components (공용 컴포넌트 및 유틸 관리)
  - halli-galli (할리갈리 게임 패키지)

🧐 테스트
yarn
yarn build
or
cd packages/chromeside-board
yarn dev

✨ 참고내용 / 관련 MR

### CODE REVIEW

**Review by jeeyeon0523 — DISMISSED** [review]

**Review by yunju-choi — DISMISSED** [review]

**Comment by yunju-choi** [comment]
chromeside-board가 build fail 나는데 저만 이럴까요??

**Comment by sprinter707** [comment]
> chromeside-board가 build fail 나는데 저만 이럴까요??

어떤 에러가 발생하나요???

**Comment by yunju-choi** [comment]
메세지는 아래와 같이 찍힙니다!

    ✖  chromeside-board:build
$ tsc && vite build
       vite v4.2.1 building for production...
       transforming...
       ✓ 5 modules transformed.
       ✓ built in 53ms
       [commonjs--resolver] Failed to resolve entry for package "ui-components". The package may have incorrect main/module/exports specified in its package.json.
       error during build:
       Error: Failed to resolve entry for package "ui-components". The package may have incorrect main/module/exports specified in its package.json.
           at packageEntryFailure (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/vite/dist/node/chunks/dep-79892de8.js:23266:11)
           at resolvePackageEntry (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/vite/dist/node/chunks/dep-79892de8.js:23263:5)
           at tryNodeResolve (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/vite/dist/node/chunks/dep-79892de8.js:23000:20)
           at Object.resolveId (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/vite/dist/node/chunks/dep-79892de8.js:22762:28)
           at file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/rollup/dist/es/shared/node-entry.js:24395:40
           at async PluginDriver.hookFirstAndGetPlugin (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/rollup/dist/es/shared/node-entry.js:24295:28)
           at async resolveId (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/rollup/dist/es/shared/node-entry.js:23192:26)
           at async ModuleLoader.resolveId (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/rollup/dist/es/shared/node-entry.js:23456:15)
           at async Object.resolveId (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/vite/dist/node/chunks/dep-79892de8.js:8006:10)
           at async PluginDriver.hookFirstAndGetPlugin (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/rollup/dist/es/shared/node-entry.js:24295:28)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
       
    ✔  ui-components:build (2s)
    ✔  halli-galli:build (2s)

 —————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  Lerna (powered by Nx)   Ran target build for 3 projects (2s)
 
    ✔    2/3 succeeded [0 read from cache]
 
    ✖    1/3 targets failed, including the following:
         - chromeside-board:build

**Comment by sprinter707** [comment]
> 메세지는 아래와 같이 찍힙니다!
> 
> ```
> ✖  chromeside-board:build
> ```
> 
> $ tsc && vite build vite v4.2.1 building for production... transforming... ✓ 5 modules transformed. ✓ built in 53ms [commonjs--resolver] Failed to resolve entry for package "ui-components". The package may have incorrect main/module/exports specified in its package.json. error during build: Error: Failed to resolve entry for package "ui-components". The package may have incorrect main/module/exports specified in its package.json. at packageEntryFailure (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/vite/dist/node/chunks/dep-79892de8.js:23266:11) at resolvePackageEntry (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/vite/dist/node/chunks/dep-79892de8.js:23263:5) at tryNodeResolve (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/vite/dist/node/chunks/dep-79892de8.js:23000:20) at Object.resolveId (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/vite/dist/node/chunks/dep-79892de8.js:22762:28) at file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/rollup/dist/es/shared/node-entry.js:24395:40 at async PluginDriver.hookFirstAndGetPlugin (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/rollup/dist/es/shared/node-entry.js:24295:28) at async resolveId (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/rollup/dist/es/shared/node-entry.js:23192:26) at async ModuleLoader.resolveId (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/rollup/dist/es/shared/node-entry.js:23456:15) at async Object.resolveId (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/vite/dist/node/chunks/dep-79892de8.js:8006:10) at async PluginDriver.hookFirstAndGetPlugin (file:///Users/cherry/Desktop/project/CN-SIDE-1/node_modules/rollup/dist/es/shared/node-entry.js:24295:28) error Command failed with exit code 1. info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
> 
> ```
> ✔  ui-components:build (2s)
> ✔  halli-galli:build (2s)
> ```
> 
> —————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
> 
> > Lerna (powered by Nx)   Ran target build for 3 projects (2s)
> 
> ```
> ✔    2/3 succeeded [0 read from cache]
> 
> ✖    1/3 targets failed, including the following:
>      - chromeside-board:build
> ```

웁스... yarn dev에서는 정상동작해서 문제 없는걸로 생각했네요... 저도 같은 증상 나옵니다. 
확인해서 수정할께요

**Comment by sprinter707** [comment]
@ga0hyeon 빌드에러 확인해보니 tsconfig.json에서 빌드시 다른 패키지로 export하기 위해서 만들어둔 index.ts 파일에서 에러가 발생했던거 같아요. 현재는 빼버렸던 파일이라서 다시 tsconfig.json을 원래대로 돌려서 빌드 성공확인했어요

**Comment by sprinter707** [comment]
@yunju-choi 
빌드에러 확인해보니 tsconfig.json에서 빌드시 다른 패키지로 export하기 위해서 만들어둔 index.ts 파일에서 에러가 발생했던거 같아요. 현재는 빼버렸던 파일이라서 다시 tsconfig.json을 원래대로 돌려서 빌드 성공확인했어요

File: README.md [hunk 1]
```diff
@@ -4,10 +4,16 @@
 >
 > > tech : monorepo, (React, zustand, storybook, tailwind), NodeJS(Express, Socket.io)
 
+## packages
+
+- chromeside-board (메인 로비)
```
Comment (ga0hyeon) [inline README.md:9]: .storybook 폴더가 chromeside-board 폴더 하위에 있는게 맞는걸까요? ui-components 밑에 있을거라고 생각했습니다.

Comment (sprinter707) [reply README.md:9]: chromeside-board 이전 패키지 명에서 테스트로 사용하다가 흔적이 남아있엇어요
지금은 제거했습니다.

**Review by yunju-choi — APPROVED** [review]

**Review by jeeyeon0523 — APPROVED** [review]

**Pull request merged**